### PR TITLE
Address the vulnerability reported in Security Wave II

### PR DESCRIPTION
**Pull request by nwnt**
### Which issue this PR addresses:

Fixes: ARO-9396

### What this PR does / why we need it:
To address the main CVE listed for Security Wave II.  Docker v24.0.9 has a critical vulnerability.

### Test plan for issue:
This is just upgrading the docker lib version.  E2E should suffice.

### Is there any documentation that needs to be updated for this PR?

No.

### How do you know this will function as expected in production? 

If it builds successfully, it works.

### CODE REVIEW

**Comment by SudoBrendan** [comment]
Just a side note: when we merge this - we need to send out a team-wide communication on this change, everyone will need to bump their local `podman` version to 5.0.3 - it doesn't look like we document the version in https://github.com/Azure/ARO-RP/blob/loki/pipeline-swarm/docs/prepare-your-dev-environment.md so I don't think this needs a doc change, but this will affect things like local cluster creations in dev environments.

File: .pipelines/ci.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ pr:
 resources:
   containers:
     - container: golang
-      image: mcr.microsoft.com/oss/go/microsoft/golang:1.21.13-1-cbl-mariner2.0-amd64
+      image: mcr.microsoft.com/onebranch/cbl-mariner/build:2.0
```
[the remark this answers is not in the document]
Comment (nwnt) [reply .pipelines/ci.yml:26]: Noted.  This is not the final version btw, just trying out and see if I could get the pipeline to work.

**Comment by nwnt** [comment]
azp run ci,e2e

File: .pipelines/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nwnt) [reply .pipelines/ci.yml:26]: I couldn't use `mcr.microsoft.com/oss/go/microsoft/golang:1.21.13-1-cbl-mariner2.0-amd64` and had to stick with onebranch build image as the former doesn't allow tdnf install the required packages for podman 5 build kicked off by `go generate`.  Instead I pinned the version of go during the installation this that should address this concern then.

**Review by hlipsig — APPROVED** [review]
LGTM

**Pull request merged**